transport/tchannel/handler.go

```diff
@@ -170,6 +172,16 @@ func (rw *responseWriter) AddHeaders(h transport.Headers) {
 	}
 }
 
+func (rw *responseWriter) SetApplicationError() {
+	if rw.wroteHeaders {
+		panic("SetApplicationError() cannot be called after calling Write().")
```

Comment:
Not to my knowledge. `log.Fatal` will exit the program with no chance of stopping it. `panic`s can be rescued.